### ISSUE TITLE
Fix deprecation-, unused- and other warnings (from Eclipse) in JGraphT's tests

### ISSUE THREAD

**Issue by HannesWell** [jgrapht/jgrapht#1056]
```
 * JGraphT version: 1.5.1
```

**Issue**
The Problems View of Eclipse shows me 302 warnings in the test code of JGraphT.
Most warnings are issued due to usage of deprecated assertion methods from JUnit and because of unused local variables or methods.
I suggest to fix those warnings.

The deprecation warnings can be fixed by using the recommended new methods or constructs.
The unused warnings can be either fixed by removing the variable declaration/assignment or by adding a `@SuppressWarnings("unused")` annotation where removal of an assignment would disturb the code-structure of the test. 

**Other information**
I can prepare a PR if you agree.

**Comment by jsichi** [comment]
Yes, please go ahead.  We don't have a way to enforce these, so this is a Sisyphean task.  :)

**Comment by HannesWell** [comment]
Indeed.
But to avoid those warnings the Java-Compiler can emit, _failOnWarning_ and _showWarnings_ could also be enabled for compiling the test-sources in the root pom.xml.

**Comment by HannesWell** [comment]
Besides enabling compiler warnings for tests too, the static code analyzer tool `PMD` could be used to detect unused code:
https://pmd.github.io/latest/pmd_rules_java.html (search the site for unused)
If you think that using PMD besides checktyle could be an option for jgrapht, I can evaluate it. 
What I found so far is that PMD obeys the `@SuppressWarnings("unused")` used by Eclipse (and maybe other IDEs?) to suppress unused warnings: https://pmd.github.io/latest/pmd_userdocs_suppressing_warnings.html

Alternatively there is also the static code analyzer spot-bugs (the successor of find-bugs). I have no experience with either of them but what I've read so far is that they have better capabilities in finding flaws/errors/bugs in the logic of the code compared to checkstyle while checkstyle is better in checking the style (what a surprise) of a source file. But there seems to be no general problem in using them both (some even recommend it).

Besides that I got some unused warnings for some locations in jgrapht's main code that I could fix too.

**Comment by jsichi** [comment]
As long as we don't run into too much noise as in #688, it is always helpful to be considering additional automated static analysis/enforcement.  So if you have time for evaluating the ones you mentioned with respect to JGraphT, please go ahead.
